lms/static/scripts/frontend_apps/components/dashboard/StudentsActivity.tsx

```diff
@@ -6,45 +6,44 @@ import {
 } from '@hypothesis/frontend-shared';
 import type { DataTableProps } from '@hypothesis/frontend-shared';
 import { useState } from 'preact/hooks';
+import { useParams } from 'wouter-preact';
 
-import type { StudentStats } from '../../api-types';
+import type { Assignment, StudentStats } from '../../api-types';
+import { useConfig } from '../../config';
+import { useAPIFetch } from '../../utils/api';
 import { formatDateTime } from '../../utils/date';
 
-export type AssignmentInfo = {
-  title: string;
-};
-
-export type StudentsActivityTableProps = {
-  assignment: AssignmentInfo;
-  students: StudentStats[];
-  loading?: boolean;
-};
-
 type MandatoryOrder<T> = NonNullable<DataTableProps<T>['order']>;
 
-export default function StudentsActivityTable({
-  assignment,
-  students,
-  loading,
-}: StudentsActivityTableProps) {
-  const title = `Assignment: ${assignment.title}`;
+export default function StudentsActivity() {
+  const { dashboard } = useConfig(['dashboard']);
+  const { routes } = dashboard;
+  const { assignmentId } = useParams<{ assignmentId: string }>();
+  const assignment = useAPIFetch<Assignment>(
+    routes.assignment.replace(':assignment_id', assignmentId),
+  );
+  const students = useAPIFetch<StudentStats[]>(
+    routes.assignment_stats.replace(':assignment_id', assignmentId),
```

Comment:
We should add a general mechanism for this in `useAPIFetch`.

Reply:
I forgot to bring some comments from the old PR that was superseded by this one. In short, yes, my plan was to bring the logic we have in client, but I like the idea of wrapping that inside `useAPIFetch`/`apiCall`.

Reply:
To clarify, I meant that my plan was to add the general logic in a follow-up PR.